app/codes/committeemanager.py

```diff
@@ -28,22 +29,30 @@ def get_number_from_hash(block_hash):
     return ord(block_hash[0])
 
 
-def weighted_random_choices(population, weights, k, seed_prefix):
+def weighted_random_choices(population, weights, k, seed_prefix=0):
     if len(population) < k:
         raise Exception('Population less than selection count')
     
-    selections = []
-    previous_idx = 0
-    while len(selections) < k:
-        random.seed(seed_prefix + previous_idx)
-        choice = random.choices(population, weights=weights)[0]
-        index = population.index(choice)
-        previous_idx = index
-        selections.append(choice)
-        del population[index]
-        del weights[index]
+    v = [random.random() ** (1 / w) for w in weights]
```

Comment:
Shouldn't this be using the seed_prefix?

Reply:
yes, that was missed. correcting.